cms/core/utils.py

```diff
@@ -75,3 +81,17 @@ def latex_formula_to_svg(latex: str, *, fontsize: int = 18, transparent: bool =
         svg_string = "\n".join(svg_string.split("\n")[3:])
 
     return svg_string
+
+
+def redirect_to_parent_listing(
+    page: "Page", request: "HttpRequest", get_listing_url_method: str, redirect_status: int = 307
+) -> HttpResponseRedirect:
+    """Redirects to the parent page's listing URL if available, otherwise to the parent page itself."""
+    parent = page.get_parent().specific_deferred
```

Comment:
Since these are generic utils functions, we can't assume the parent exists, the parent may return none, so let's safeguard against this.

```py
parent = getattr(page.get_parent(), "specific_deferred", None)
if not parent:
    return redirect("/")
```

Reply:
Done! [93adbdb](https://github.com/ONSdigital/dis-wagtail/pull/387/commits/93adbdb91f81592ff7116d895167b95be08b8e7c)